docs/content/preview/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -182,6 +182,18 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+
+{{< warning >}}
+Adding a `PRIMARY KEY` constraint will result in a full table rewrite and full-rewrite of all indexes associated with the table.
+This happens because of the clustered storage by primary-key that YugabyteDB uses to store rows and indexes. 
+Tables without a `PRIMARY KEY` have a hidden one underneath and rows are stored clustered on it. The secondary indexes of the table
+link to this hidden `PRIMARY KEY`.
+During this time, you may lose modification made to the table. 
```

Comment:
'this time' - what time is that? While the tables and indexes are being rewritten?

While the tables and indexes are being rewritten, you may lose any modifications made to the table.

Reply:
Yes "While the tables and indexes are being rewritten, you may lose any modifications made to the table.".